test/e2e/cluster_autoscaling.go

```diff
@@ -0,0 +1,190 @@
+			clusterParams.ClusterName = customerClusterName
+			clusterParams.ManagedResourceGroupName = managedResourceGroupName
+			clusterParams.NsgResourceID = nsgResourceID
+			clusterParams.SubnetResourceID = vnetSubnetResourceID
+			clusterParams.VnetName = customerVnetName
+			clusterParams.UserAssignedIdentitiesProfile = userAssignedIdentitiesProfile
+			clusterParams.Identity = identityProfile
+			clusterParams.KeyVaultName = keyVaultNameStr
+			clusterParams.EtcdEncryptionKeyName = etcdEncryptionKeyName
+			clusterParams.EtcdEncryptionKeyVersion = etcdEncryptionKeyVersion
+			clusterParams.Autoscaling = &hcpsdk20240610preview.ClusterAutoscalingProfile{
+				MaxNodeProvisionTimeSeconds: to.Ptr(autoscalingMaxNodeProvisionTimeSeconds),
+				MaxPodGracePeriodSeconds:    to.Ptr(autoscalingMaxPodGracePeriodSeconds),
+				PodPriorityThreshold:        to.Ptr(autoscalingPodPriorityThreshold),
+			}
+
+			err = framework.CreateHCPClusterFromParam(ctx,
+				tc,
+				*resourceGroup.Name,
+				clusterParams,
+				45*time.Minute,
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("ensuring the custom autoscaling was honored")
+			got, err := framework.GetHCPCluster(
+				ctx,
+				tc.Get20240610ClientFactoryOrDie(ctx).NewHcpOpenShiftClustersClient(),
+				*resourceGroup.Name,
+				customerClusterName,
+			)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(got.Properties.Autoscaling).ToNot(BeNil())
+			Expect(got.Properties.Autoscaling.MaxNodeProvisionTimeSeconds).To(Equal(to.Ptr(autoscalingMaxNodeProvisionTimeSeconds)))
+			Expect(got.Properties.Autoscaling.MaxPodGracePeriodSeconds).To(Equal(to.Ptr(autoscalingMaxPodGracePeriodSeconds)))
+			Expect(got.Properties.Autoscaling.PodPriorityThreshold).To(Equal(to.Ptr(autoscalingPodPriorityThreshold)))
+
+			By("creating the node pool")
+			nodePoolParams := framework.NewDefaultNodePoolParams()
+			nodePoolParams.ClusterName = customerClusterName
+			nodePoolParams.NodePoolName = customerNodePoolName
+			nodePoolParams.Replicas = int32(2)
+
+			err = framework.CreateNodePoolFromParam(ctx,
+				tc,
+				*resourceGroup.Name,
+				customerClusterName,
+				nodePoolParams,
+				45*time.Minute,
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("patching the cluster to set maxNodesTotal")
+			update, err := framework.UpdateHCPCluster(
+				ctx,
+				tc.Get20240610ClientFactoryOrDie(ctx).NewHcpOpenShiftClustersClient(),
+				*resourceGroup.Name,
+				customerClusterName,
+				hcpsdk20240610preview.HcpOpenShiftClusterUpdate{
+					Properties: &hcpsdk20240610preview.HcpOpenShiftClusterPropertiesUpdate{
+						Autoscaling: &hcpsdk20240610preview.ClusterAutoscalingProfile{
+							MaxNodesTotal: to.Ptr(autoscalingMaxNodesTotal),
+						},
+					},
+				},
+				5*time.Minute,
+			)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(got.Properties.Autoscaling).ToNot(BeNil())
+			Expect(update.Properties.Autoscaling.MaxNodesTotal).To(Equal(to.Ptr(autoscalingMaxNodesTotal)))
```

Comment:
Did we expect the e2e to actually autoscale? Simply verifying the API is a start, but I think the real user intent here is to have an auto-scaling node pool, not just to have their intent saved in a database :)

Reply:
I haven't been able to induce any deterministic autoscaling behavior in my test environment by fiddling with node limits.  Not sure if I'm doing something wrong or if it's not fully wired up further down the stack.

Anyway, that can be a followup.  The scope of this pull request is forwarding autoscaling updates over to Cluster Service correctly.